src/common/framework/test_group.ts

```diff
@@ -107,36 +108,36 @@ class TestGroup<F extends Fixture> implements TestGroupBuilder<F> {
   }
 }
 
-interface TestBuilderWithName<F extends Fixture> extends TestBuilderWithParams<F, {}> {
+interface TestBuilderWithName<F extends Fixture> extends TestBuilderWithCases<F, {}> {
   desc(description: string): this;
   /** @deprecated use cases() and/or subcases() instead */
-  params<NewP extends {}>(specs: Iterable<NewP>): TestBuilderWithParams<F, NewP>;
-  cases<NewP extends {}>(specs: Iterable<NewP>): TestBuilderWithParams<F, NewP>;
+  params<NewP extends {}>(specs: Iterable<NewP>): TestBuilderWithSubcases<F, NewP>;
+  cases<NewP extends {}>(specs: Iterable<NewP>): TestBuilderWithCases<F, NewP>;
```

Comment:
Shouldn't these return the same type? I only expect it to return TestBuilderWithSubcases when you call .subcases 

or maybe it's because if you're using deprecated .params, you shouldn't use the new .subcases

Reply:
> or maybe it's because if you're using deprecated .params, you shouldn't use the new .subcases

yeah, it's not necessary to do it this way, but this is why.